boards/DCM/Test/Src/torquevectoring/Test_ActiveDifferential.cpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * Notes:
+ * extern "C": specifies C linkage for functions to compiler
+ *
+ *
+ *
+*/
+
+#include "Test_Dcm.h"
+#include "App_SharedMacros.h"
+
+extern "C"
+{
+#include "torquevectoring/App_PowerLimiting.h"
+#include "torquevectoring/App_ActiveDifferential.h"
+#include "torquevectoring/App_TorqueVectoringConstants.h"
+#include "torquevectoring/App_TorqueVectoring.h"
+#include "math.h"
+}
+/**
+ * Define SetUp and TearDown to be called before every TEST_F call
+*/
+class ActiveDifferentialTest: public testing::Test
+{
+protected:
+    //TODO: Find out if this is necessary if no IO functions are called.
+    void SetUp() override
+    {
+        return;
+    }
+    void TearDown() override
+    {
+        return;
+    }
+};
+
+TEST_F(ActiveDifferentialTest, max_power){
+    struct ActiveDifferential_Inputs active_diff_inputs_t1;
+    struct ActiveDifferential_Outputs actual_active_diff_outputs_t1;
+    float wheel_angle_t1 = 30.0;
+    float wheel_speed_left_rpm_t1 = 135, wheel_speed_right_rpm_t1 = 135; //135 rpm corresponds to about 30 mph
+    active_diff_inputs_t1 = {wheel_angle_t1, wheel_speed_left_rpm_t1, wheel_speed_right_rpm_t1, POWER_LIMIT_CAR_kW};
+    App_ActiveDifferential_ComputeTorque(&active_diff_inputs_t1, &actual_active_diff_outputs_t1);
+    float delta_t1 = TRACK_WIDTH_mm* tanf(DEG_TO_RAD(wheel_angle_t1))/(2 * WHEELBASE_mm);
+    float cl_t1 = 1 + delta_t1;
+    float cr_t1 = 1 - delta_t1;
+    float torque_t1 = (POWER_TO_TORQUE_CONVERSION_FACTOR * POWER_LIMIT_CAR_kW)/(wheel_speed_left_rpm_t1*cl_t1 + wheel_speed_right_rpm_t1*cr_t1 + SMALL_EPSILON);
+    float expected_torque_left_t1 = cl_t1*torque_t1;
+    float expected_torque_right_t1 = cr_t1*torque_t1;
+    float max_torque_t1 = fmaxf(expected_torque_left_t1, expected_torque_right_t1);
+    if(max_torque_t1 > MOTOR_TORQUE_LIMIT_Nm){
+        float sigma_t1 = MOTOR_TORQUE_LIMIT_Nm/max_torque_t1;
+        expected_torque_right_t1 = sigma_t1*expected_torque_right_t1;
+        expected_torque_left_t1 = sigma_t1*expected_torque_left_t1;
+    }
+    float expected_ratio_t1 = cl_t1/cr_t1;
+    float calculated_ratio_t1 = expected_torque_left_t1/expected_torque_right_t1;
+    float actual_ratio_t1 = actual_active_diff_outputs_t1.torque_left_Nm/actual_active_diff_outputs_t1.torque_right_Nm;
+    ASSERT_FLOAT_EQ(expected_torque_left_t1, actual_active_diff_outputs_t1.torque_left_Nm);
+    ASSERT_FLOAT_EQ(expected_torque_right_t1, actual_active_diff_outputs_t1.torque_right_Nm);
+    ASSERT_TRUE(actual_active_diff_outputs_t1.torque_right_Nm <= MOTOR_TORQUE_LIMIT_Nm);
+    ASSERT_TRUE(actual_active_diff_outputs_t1.torque_left_Nm <= MOTOR_TORQUE_LIMIT_Nm);
+    ASSERT_FLOAT_EQ(expected_ratio_t1,calculated_ratio_t1);
+    ASSERT_FLOAT_EQ(expected_ratio_t1, actual_ratio_t1);
+};
```

Comment:
These could definitely be more readable
Maybe break this into sections + add more comments 
Right now its hard to tell what is actually being tested
Also any common setup/teardown code can go in SetUp and TearDown above, so any values used in all the tests can be declared as class variables and set in SetUp

Reply:
@1mackenziekyle